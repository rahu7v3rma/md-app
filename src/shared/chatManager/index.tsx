import React, { FunctionComponent, useState } from 'react';

import { AppChatProvider } from '@/contexts/appChat';

import { useChatNotifications, useInitChatClient } from './hooks';

type Props = {
    children?: React.ReactNode;
};

const ChatManager: FunctionComponent<Props> = ({ children }: Props) => {
    const { chatClient } = useInitChatClient();

    const [activeChatChannel, setActiveChatChannel] = useState<string | null>(
        null
    );

    useChatNotifications(chatClient, activeChatChannel);

    return (
        <AppChatProvider value={{ chatClient, setActiveChatChannel }}>
            {children}
        </AppChatProvider>
    );
};

export default ChatManager;
