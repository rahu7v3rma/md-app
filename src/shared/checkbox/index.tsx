import React, { FunctionComponent } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

import { CheckIcon } from '@/assets/svgs';
import { Colors } from '@/theme/colors';

type Props = {
    checked: boolean;
    onChange: (value: boolean) => void;
};

const Checkbox: FunctionComponent<Props> = ({ checked, onChange }: Props) => {
    return (
        <TouchableOpacity
            style={[checkboxStyle.wrapper, checked && checkboxStyle.checked]}
            activeOpacity={0.8}
            onPress={() => onChange(!checked)}
        >
            {checked && (
                <CheckIcon color={Colors.extras.white} width={18} height={18} />
            )}
        </TouchableOpacity>
    );
};

export default Checkbox;

const checkboxStyle = StyleSheet.create({
    wrapper: {
        backgroundColor: Colors.text.white,
        borderWidth: 1.5,
        borderColor: Colors.button.app_button_disabled_border,
        width: 28,
        height: 28,
        borderRadius: 9,
        alignItems: 'center',
        justifyContent: 'center'
    },
    checked: {
        backgroundColor: Colors.text.green,
        borderWidth: 0
    }
});
