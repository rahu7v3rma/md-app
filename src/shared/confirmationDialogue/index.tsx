import React, { FunctionComponent, RefObject } from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';

import Button from '@/shared/button';
import Text, { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';

type Props = {
    bottomSheetRef: RefObject<RBSheet>;
    title: string;
    subTitle?: string;
    dismissBtnTitle?: string;
    confirmBtnTitle?: string;
    onDismissBtnHandler?: () => void;
    onConfirmBtnHandler?: () => void;
    confirmBtnStyles?: StyleProp<ViewStyle>;
};

const ConfirmationDialogue: FunctionComponent<Props> = ({
    bottomSheetRef,
    title,
    subTitle,
    dismissBtnTitle,
    confirmBtnTitle,
    onDismissBtnHandler,
    onConfirmBtnHandler,
    confirmBtnStyles
}: Props) => {
    return (
        <RBSheet
            ref={bottomSheetRef}
            closeOnDragDown={true}
            animationType={'fade'}
            closeOnPressMask={true}
            customStyles={{
                wrapper: styles.wrapper,
                container: styles.contentContainer,
                draggableIcon: styles.draggableIcon
            }}
        >
            <View style={styles.titlesView}>
                <Text
                    size={Size.Large}
                    fontWeight="bold"
                    color={Colors.text.black_gray}
                    style={styles.title}
                >
                    {title}
                </Text>
                {subTitle && (
                    <Text
                        testID="confirmationDialogueSubTitle"
                        size={Size.XXSmall}
                        color={Colors.text.black}
                        style={styles.subTitle}
                    >
                        {subTitle}
                    </Text>
                )}
            </View>
            <View style={styles.buttonsView}>
                <Button
                    block={true}
                    bordered={false}
                    style={styles.firstButton}
                    onPress={onDismissBtnHandler}
                    testID="dismissBtn"
                >
                    <Text color={Colors.text.primary} fontWeight="600">
                        {dismissBtnTitle}
                    </Text>
                </Button>
                <Button
                    block={true}
                    bordered={false}
                    style={[styles.secondButton, confirmBtnStyles]}
                    onPress={onConfirmBtnHandler}
                    testID="confirmBtn"
                >
                    <Text color={Colors.text.white} fontWeight="600">
                        {confirmBtnTitle}
                    </Text>
                </Button>
            </View>
        </RBSheet>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        backgroundColor: Colors.theme.app_sheet_background_color
    },
    contentContainer: {
        height: 'auto',
        opacity: 1,
        backgroundColor: Colors.extras.white,
        alignItems: 'center',
        borderTopLeftRadius: 40,
        borderTopRightRadius: 40
    },
    draggableIcon: {
        width: 0
    },
    titlesView: {
        width: '80%',
        marginTop: 20
    },
    title: {
        textAlign: 'center'
    },
    subTitle: {
        paddingTop: 20,
        textAlign: 'center'
    },
    buttonsView: {
        width: '100%',
        paddingHorizontal: 20,
        marginTop: 41.86,
        marginBottom: 48
    },
    firstButton: {
        height: 56,
        paddingVertical: 0,
        backgroundColor: Colors.theme.primary_light,
        marginBottom: 16.5
    },
    secondButton: {
        height: 56,
        paddingVertical: 0,
        backgroundColor: Colors.theme.primary,
        borderColor: Colors.button.app_button_primary_border
    }
});

export default ConfirmationDialogue;
