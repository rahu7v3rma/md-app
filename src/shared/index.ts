import Button from './button';
import ChatAvatar from './chatAvatar';
import ChatListItem from './chatListItem';
import ChatManager from './chatManager';
import Checkbox from './checkbox';
import ConfirmationDialogue from './confirmationDialogue';
import CustomCacheImage from './customCacheImage';
import CustomStatusBar from './customStatusBar';
import Header from './header';
import ImageChooser from './imageChooser';
import Input from './input';
import LessonCard from './lessonCard';
import ListItem from './listItem';
import LogBookListItem from './logBookListItem';
import LogInputDropdown from './logInputDropdown';
import LogTab from './logTab';
import LogTimePicker from './logTimePicker';
import LogUnitPicker from './logUnitPicker';
import MainTabBar from './mainTabBar';
import NoDataAvailable from './noDataAvailable';
import PdfViewer from './pdfViewer';
import PlatformImage from './platformImage';
import ProfileImage from './profileImage';
import Progressbar from './progressBar';
import ProgressCircle from './progressCircle';
import Switch from './switch';
import Text from './text';
import VideoPlayer from './videoPlayer';
import Wrapper from './wrapper';

export {
    Button,
    ChatAvatar,
    ChatListItem,
    ChatManager,
    Checkbox,
    ConfirmationDialogue,
    CustomCacheImage,
    CustomStatusBar,
    Header,
    ImageChooser,
    Input,
    LessonCard,
    ListItem,
    LogBookListItem,
    LogInputDropdown,
    LogTab,
    LogTimePicker,
    LogUnitPicker,
    MainTabBar,
    NoDataAvailable,
    PlatformImage,
    ProfileImage,
    Progressbar,
    ProgressCircle,
    Switch,
    Text,
    VideoPlayer,
    Wrapper,
    PdfViewer
};
