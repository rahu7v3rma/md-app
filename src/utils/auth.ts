import { CommonActions } from '@react-navigation/native';
import * as Keychain from 'react-native-keychain';

import { navigationRef } from '@/navigation';
import { resetNotifications } from '@/services/notification';
import { resetStore } from '@/store';

export const setAuthToken = (token: string) => {
    Keychain.setGenericPassword('token', token);
};

export const getAuthToken = (): Promise<string | null> => {
    return Keychain.getGenericPassword().then((credentials) =>
        credentials === false ? null : credentials.password
    );
};

export const AUTHORIZATION_HEADER_NAME = 'X-Authorization';

export const getAuthorizationHeaderValue = () => {
    return getAuthToken()
        .then((authToken) => `Token ${authToken || ''}`)
        .catch(() => undefined);
};

export const resetAuthToken = () => {
    Keychain.resetGenericPassword();
};

export const logoutAction = (skipNavigation: boolean = false) => {
    resetStore();
    resetNotifications();
    resetAuthToken();

    if (!skipNavigation) {
        navigationRef.dispatch(
            CommonActions.reset({
                index: 0,
                routes: [{ name: 'SignIn' }]
            })
        );
    }
};
