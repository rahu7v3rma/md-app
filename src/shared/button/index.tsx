import React, { FunctionComponent } from 'react';
import {
    GestureResponderEvent,
    StyleProp,
    StyleSheet,
    TouchableOpacity,
    ViewStyle
} from 'react-native';

import { Colors } from '@/theme/colors';
import { COMMON } from '@/utils/common';

type Props = {
    children?: React.ReactNode;
    primary?: boolean;
    bordered?: boolean;
    rounded?: boolean;
    block?: boolean;
    style?: StyleProp<ViewStyle>;
    disabled?: boolean;
    onPress?: (e: GestureResponderEvent) => void;
    testID?: string;
};

const Button: FunctionComponent<Props> = ({
    children,
    primary = false,
    bordered = true,
    rounded,
    block,
    style,
    disabled,
    onPress,
    testID
}: Props) => {
    return (
        <TouchableOpacity
            testID={testID}
            style={[
                buttonStyle.wrapper,
                bordered && buttonStyle.bordered,
                primary && buttonStyle.primary,
                disabled && buttonStyle.disabled,
                rounded && buttonStyle.rounded,
                block && buttonStyle.block,
                style
            ]}
            disabled={disabled}
            activeOpacity={disabled ? 1 : 0.75}
            onPress={onPress}
        >
            {children}
        </TouchableOpacity>
    );
};

export default Button;

const buttonStyle = StyleSheet.create({
    wrapper: {
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: 15.5,
        paddingVertical: COMMON.isIos ? 24 : 17.5,
        borderRadius: 20,
        margin: 0
    },
    bordered: {
        borderWidth: 1,
        borderBottomWidth: 5,
        borderColor: Colors.theme.primary
    },
    primary: {
        backgroundColor: Colors.theme.primary,
        borderColor: Colors.theme.primary
    },
    disabled: {
        backgroundColor: Colors.button.app_button_disabled_bg,
        borderColor: Colors.button.app_button_disabled_border
    },
    rounded: {
        borderRadius: 100,
        backgroundColor: Colors.button.app_btn_light_color
    },
    block: {
        width: '100%'
    }
});
