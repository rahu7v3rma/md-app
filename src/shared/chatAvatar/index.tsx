import React, { FunctionComponent } from 'react';
import { Image, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';

import DefaultAvatar from '@/assets/images/default_avatar.jpg';
import { PlatformImage } from '@/shared';
import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
    path?: string;
    online?: boolean;
    style?: StyleProp<ViewStyle>;
};

const ChatAvatar: FunctionComponent<Props> = ({
    width,
    height,
    path,
    online,
    style
}: Props) => {
    const borderRadius = width ? width / 2 : 0;

    return (
        <>
            <View style={[styles.wrapper, style]}>
                {path ? (
                    <PlatformImage
                        style={{ borderRadius, width, height }}
                        imageId={path}
                        width={width}
                        height={height}
                    />
                ) : (
                    <Image
                        style={{ borderRadius, width, height }}
                        source={DefaultAvatar}
                    />
                )}
                {online && <View testID="online" style={styles.onlineDot} />}
            </View>
        </>
    );
};

export default ChatAvatar;

const styles = StyleSheet.create({
    wrapper: {
        justifyContent: 'center'
    },
    onlineDot: {
        backgroundColor: Colors.icons.green_check_mark,
        borderStyle: 'solid',
        borderColor: Colors.extras.white,
        width: 12,
        borderWidth: 2,
        height: 12,
        borderRadius: 12,
        zIndex: 1,
        position: 'absolute',
        top: 0,
        transform: [{ translateY: -3 }, { translateX: -3 }]
    }
});
