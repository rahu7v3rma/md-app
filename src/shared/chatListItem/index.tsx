import React, { FunctionComponent } from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { ArrowNext } from '@/assets/svgs';
import { ChatAvatar, Text } from '@/shared';
import { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';
import { LayoutContants } from '@/theme/layoutContants';

type Props = {
    avatar?: any;
    label: string;
    lastMessage?: string;
    time?: string;
    newMessagesCount?: number;
    isOnline?: boolean;
    onPress?: () => void;
    style?: any;
    mainTouchID?: string;
    msgCountID?: string;
};

const ChatListItem: FunctionComponent<Props> = ({
    avatar,
    isOnline = false,
    label,
    lastMessage = '',
    time = '',
    newMessagesCount = 0,
    onPress,
    style,
    mainTouchID,
    msgCountID
}: Props) => {
    return (
        <TouchableOpacity
            testID={mainTouchID}
            onPress={onPress}
            style={[styles.wrapper, style]}
        >
            <View style={styles.firstCol}>
                <View style={styles.avatarView}>
                    <ChatAvatar
                        path={avatar}
                        width={24}
                        height={24}
                        online={isOnline}
                        style={styles.avatar}
                    />
                </View>
            </View>
            <View style={styles.secondCol}>
                <Text
                    size={Size.XSmall}
                    fontWeight="700"
                    color={Colors.text.black}
                >
                    {label}
                </Text>
                {lastMessage !== '' && (
                    <Text size={Size.XXXSmall} color={Colors.text.gray}>
                        {lastMessage.substring(0, 29).concat('...')}
                    </Text>
                )}
            </View>
            <View style={styles.thirdCol}>
                {!time ? (
                    <ArrowNext
                        width={32}
                        height={32}
                        color={Colors.text.purple}
                    />
                ) : (
                    <>
                        <Text size={Size.XXXSmall} color={Colors.text.gray}>
                            {time}
                        </Text>
                        {newMessagesCount > 0 && (
                            <View
                                testID={msgCountID}
                                style={styles.messagesCountView}
                            >
                                <Text color={Colors.text.white}>3</Text>
                            </View>
                        )}
                    </>
                )}
            </View>
        </TouchableOpacity>
    );
};

export default ChatListItem;

const styles = StyleSheet.create({
    wrapper: {
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: Colors.extras.white,
        borderRadius: 20,
        borderColor: Colors.button.app_button_disabled_bg,
        height: LayoutContants.cardHeight,
        elevation: 1,
        shadowOpacity: 0.06,
        shadowRadius: 3,
        shadowOffset: { width: 0, height: 0 },
        shadowColor: Colors.theme.app_sheet_background_color
    },
    firstCol: {
        width: '20%',
        alignItems: 'center'
    },
    avatarView: {
        position: 'relative',
        width: 32,
        height: 32,
        borderRadius: 32 / 2
    },
    avatar: {
        width: 32,
        height: 32
    },
    online: {
        position: 'absolute'
    },
    secondCol: {
        width: '60%'
    },
    label: {
        color: Colors.text.black
    },
    thirdCol: {
        width: '20%',
        alignItems: 'center'
    },
    messagesCountView: {
        width: 16,
        height: 16,
        borderRadius: 16 / 2,
        backgroundColor: Colors.theme.primary,
        alignItems: 'center',
        marginTop: 3
    }
});
