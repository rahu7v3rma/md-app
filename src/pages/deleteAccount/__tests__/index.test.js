import { configureStore, createSlice } from '@reduxjs/toolkit';
import React from 'react';
import { Text } from 'react-native';
import Toast from 'react-native-toast-message';
import { Provider } from 'react-redux';
import { act, create } from 'react-test-renderer';

import {
    mockDeleteAccount,
    mockDispatch,
    mockGoBack,
    mockNavigationDispatch,
    mockUnwrap
} from '@/jestSetup';
import DeleteAccount from '@/pages/deleteAccount';
import { Header, Input } from '@/shared';
import * as auth from '@/utils/auth';

const tree = create(
    <Provider
        store={configureStore({
            reducer: createSlice({ name: 'mock' }).reducer
        })}
    >
        <DeleteAccount />
    </Provider>
);

test('snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();
});

test('back button', () => {
    const backButton = tree.root.findByType(Header);
    act(() => backButton.props.onLeftBtnPress());
    expect(mockGoBack).toBeCalled();
});

test('cancel button', () => {
    const cancelButton = tree.root.findByProps({ testID: 'cancelButton' });
    act(() => cancelButton.props.onPress());
    expect(mockGoBack).toBeCalled();
});

test('proceed button', () => {
    let pageTitle = tree.root.findByProps({ testID: 'pageTitle' });
    let pageSubtitle = tree.root.findByProps({ testID: 'pageSubtitle' });
    expect(pageTitle.props.children).toStrictEqual(['Delete Account', false]);
    expect(pageSubtitle.props.children).toStrictEqual([
        'Deleting your account is an irreversible action that will permanently remove all your data from our system.',
        false
    ]);
    const proceedButton = tree.root.findByProps({ testID: 'proceedButton' });
    act(() => proceedButton.props.onPress());
    pageTitle = tree.root.findByProps({ testID: 'pageTitle' });
    pageSubtitle = tree.root.findByProps({ testID: 'pageSubtitle' });
    expect(pageTitle.props.children).toStrictEqual([
        false,
        'Confirm Account Deletion'
    ]);
    expect(pageSubtitle.props.children).toStrictEqual([
        false,
        'Please enter your account password to proceed with the account deletion process'
    ]);
    expect(tree.root.findAllByType(Input)).toHaveLength(1);
});

test('proceed without password', () => {
    const proceedButton = tree.root.findByProps({ testID: 'proceedButton' });
    act(() => proceedButton.props.onPress());
    expect(tree.root.findByType(Input).findByType(Text).props.children).toBe(
        'This field is required'
    );
    expect(mockDispatch).not.toBeCalled();
});

test('proceed with password success', async () => {
    const mockLogoutAction = jest.fn().mockImplementation(() => {});
    jest.spyOn(auth, 'logoutAction').mockImplementation(mockLogoutAction);
    const proceedButton = tree.root.findByProps({ testID: 'proceedButton' });
    act(() => proceedButton.props.onPress());
    act(() => tree.root.findByType(Input).props.onChangeText('validPassword'));
    await act(() => proceedButton.props.onPress());
    expect(mockDispatch).toBeCalled();
    expect(mockDeleteAccount).toBeCalledWith({ password: 'validPassword' });
    expect(mockLogoutAction).toBeCalledWith(true);
    expect(mockNavigationDispatch).toBeCalledWith({
        payload: {
            index: 1,
            routes: [{ name: 'SignIn' }, { name: 'DeleteAccountSuccess' }]
        },
        type: 'RESET'
    });
});

test('proceed with password error bad_credentials', async () => {
    mockUnwrap.mockRejectedValueOnce({ data: { code: 'bad_credentials' } });
    const proceedButton = tree.root.findByProps({ testID: 'proceedButton' });
    act(() => proceedButton.props.onPress());
    act(() =>
        tree.root.findByType(Input).props.onChangeText('invalidPassword')
    );
    await act(() => proceedButton.props.onPress());
    expect(mockDispatch).toBeCalled();
    expect(mockDeleteAccount).toBeCalledWith({ password: 'invalidPassword' });
    expect(Toast.show).toBeCalledWith({
        type: 'errorResponse',
        text1: 'Provided password is incorrect',
        position: 'bottom'
    });
});

test('proceed with password error something went wrong', async () => {
    mockUnwrap.mockRejectedValueOnce({
        data: { code: 'something_went_wrong' }
    });
    const proceedButton = tree.root.findByProps({ testID: 'proceedButton' });
    act(() => proceedButton.props.onPress());
    act(() =>
        tree.root.findByType(Input).props.onChangeText('invalidPassword')
    );
    await act(() => proceedButton.props.onPress());
    expect(mockDispatch).toBeCalled();
    expect(mockDeleteAccount).toBeCalledWith({ password: 'invalidPassword' });
    expect(Toast.show).toBeCalledWith({
        type: 'errorResponse',
        text1: 'Something went wrong! Please try again',
        position: 'bottom'
    });
});
