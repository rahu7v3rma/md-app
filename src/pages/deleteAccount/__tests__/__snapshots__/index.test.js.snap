// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#FAFAFF",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 22,
          "paddingHorizontal": 19,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "width": "20%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#D3E29F40",
              "borderRadius": 100,
              "elevation": 5,
              "height": 48,
              "justifyContent": "center",
              "margin": 0,
              "opacity": 1,
              "paddingHorizontal": 15.5,
              "paddingVertical": 24,
              "shadowColor": "",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 10,
              "width": 48,
            }
          }
          testID="leftButton"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={14}
            bbWidth={8}
            fill="none"
            focusable={false}
            height={14}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": "auto",
                  "width": "auto",
                },
                {
                  "flex": 0,
                  "height": 14,
                  "width": 8,
                },
              ]
            }
            vbHeight={14}
            vbWidth={8}
            width={8}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M5.97999 0.319993L2.76999 3.52999L0.79999 5.48999C-0.0300097 6.31999 -0.0300097 7.66999 0.79999 8.49999L5.97999 13.68C6.65999 14.36 7.81999 13.87 7.81999 12.92V7.30999V1.07999C7.81999 0.119992 6.65999 -0.360007 5.97999 0.319993Z"
                fill={4280752721}
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "width": "60%",
          }
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#72996B",
                "fontSize": 16,
                "fontWeight": "600",
                "lineHeight": 24,
              },
            ]
          }
          testID="title"
        />
      </View>
      <View
        style={
          {
            "alignItems": "flex-end",
            "width": "20%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-between",
        "paddingBottom": 15,
        "paddingHorizontal": 20,
      }
    }
  >
    <View>
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            {
              "marginTop": 20,
            },
            {
              "color": "#4D4F4D",
              "fontSize": 24,
              "fontWeight": "700",
              "lineHeight": 32,
            },
          ]
        }
        testID="pageTitle"
      >
        Delete Account
      </Text>
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            {
              "marginRight": 20,
              "marginTop": 30,
            },
            {
              "color": "#4D4F4D",
              "fontSize": 16,
              "fontWeight": undefined,
              "lineHeight": 24,
            },
          ]
        }
        testID="pageSubtitle"
      >
        Deleting your account is an irreversible action that will permanently remove all your data from our system.
      </Text>
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            {
              "marginTop": 40,
            },
            {
              "color": "#4D4F4D",
              "fontSize": 16,
              "fontWeight": "800",
              "lineHeight": 24,
            },
          ]
        }
      >
        Are you sure you want to proceed with deleting your account?
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#72996B",
            "borderColor": "#72996B",
            "borderRadius": 16,
            "height": 56,
            "justifyContent": "center",
            "margin": 0,
            "opacity": 1,
            "paddingHorizontal": 15.5,
            "paddingVertical": 0,
            "width": "47.5%",
          }
        }
        testID="cancelButton"
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 22,
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#CD4049",
            "borderColor": "#72996B",
            "borderRadius": 16,
            "height": 56,
            "justifyContent": "center",
            "margin": 0,
            "opacity": 1,
            "paddingHorizontal": 15.5,
            "paddingVertical": 0,
            "width": "47.5%",
          }
        }
        testID="proceedButton"
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 22,
              },
            ]
          }
        >
          Proceed
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
